Support for Proximity placement group and Availability zone as input for VMWare to Azure and Hyper-V to Azure replications in Site Recovery Service

vidyadharijami opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://aka.ms/SwaggerPRReview).

vidyadharijami committed:
Adding Proximity placement group and Availability zone for V2A and H2A

AutorestCI commented:
/azp run automation - sdk

azuresdkci commented:
Can one of the admins verify this patch?

vidyadharijami committed:
Fix after running prettier

AutorestCI commented:
/azp run automation - sdk

vidyadharijami commented:
@ChenTanyi Can you please review the PR ?

ChenTanyi commented:
@vidyadharijami API change need to be reviewed by ARM first.

pilor commented on specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:
```diff
@@ -11829,6 +11861,10 @@
           "description": "The selected recovery azure network Id.",
           "type": "string"
         },
+        "selectedTfoAzureNetworkId": {
```
>AzureNetworkId [](start = 20, length = 14)

why do these refer to "AzureNetworkId" instead of "VirtualNetworkId"?

vidyadharijami replied:
As we use virtual network id for on-premise machines, we refer AzureNetworkId for Azure failed over vms. Tfo is the acronym familiar to ASR users for Test Failover.

pilor commented on specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:
```diff
@@ -11966,6 +12010,14 @@
+        "targetAvailabilityZone": {
+          "description": "The target availability zone.",
+          "type": "string"
         }
```
please add at least one example to the spec showing what input to these properties looks like

vidyadharijami replied:
Examples were not updated as these new parameters are optional.

pilor commented:
>               "$ref": "#/definitions/EnableProtectionInput"

can you explain why this input is adding new properties but they are not reflected in the response to this API?

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:2688 in ebf307e. [](commit_id = ebf307e804782e07d98b9c525a61b66b4e37a448, deletion_comment = False)

pilor reviewed (changes requested):
:clock1:

vidyadharijami commented:
> > ```
> >           "$ref": "#/definitions/EnableProtectionInput"
> > ```
> 
> can you explain why this input is adding new properties but they are not reflected in the response to this API?
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:2688 in [ebf307e](https://github.com/Azure/azure-rest-api-specs/commit/ebf307e804782e07d98b9c525a61b66b4e37a448). [](commit_id = [ebf307e](https://github.com/Azure/azure-rest-api-specs/commit/ebf307e804782e07d98b9c525a61b66b4e37a448), deletion_comment = False)

We have added optional properties in provider specific properties for both input and API response. And here we have added for V2A and H2A.

vidyadharijami commented:
> I'm not seeing these new properties in the response. Can you point me to the line number that contains `"targetAvailabilityZone"` in the response definition for `PUT "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectedItems/{replicatedProtectedItemName}"`

Line number : 10487  in HyperVReplicaAzureReplicationDetails
    "targetAvailabilityZone": {
          "description": "The target availability zone.",
          "type": "string"
        },
        "targetProximityPlacementGroupId": {
          "description": "The target proximity placement group Id.",
          "type": "string"
        },
Line number 11899 :  in InMageAzureV2ReplicationDetails
    "targetAvailabilityZone": {
          "description": "The target availability zone.",
          "type": "string"
        },
        "targetProximityPlacementGroupId": {
          "description": "The target proximity placement group Id.",
          "type": "string"
        },
